source/backend/core.api/Middleware/LogResponseMiddleware.cs

```diff
@@ -62,11 +62,12 @@ private async Task LogResponse(HttpContext context)
 
             await _next(context);
 
-            context.Response.Body.Seek(0, SeekOrigin.Begin);
-            using var reader = new StreamReader(context.Response.Body);
             string body = null;
-            if (reader.BaseStream.Length < maxStreamLength)
+            if (context.Response.ContentLength < maxStreamLength)
```

Comment:
similar story here - larger responses end up persisting in memory after the request is handled in these logging processes.